DEVELOPER.md

```diff
@@ -0,0 +1,374 @@
+cd bottlerocket-test-system
+export TESTSYS_DIR=$(pwd)
+```
+
+The TestSys directory is now set up.
+
+### Installing tools
+
+#### Install `Kind`
+Instead of creating a long-standing eks cluster for TestSys development, `kind` enables the quick creation and destruction of local clusters that exist within the developers Docker which helps improve the testing experience.
+Follow the [installation instructions](https://kind.sigs.k8s.io/docs/user/quick-start/) to set up `kind`.
+
+#### Install `Kubectl`
+While all testing user interactions can be done with `cli`, it is helpful to look at test, resource, and pod objects in a k8s cluster with `kubectl`.
+Follow the [installation instructions]( https://kubernetes.io/docs/tasks/tools/) to set up `kubectl`.
+
+#### Install Docker
+Docker is required for building TestSys images if docker is not installed follow the [installation instructions](https://docs.docker.com/get-docker/).
+
+## Running a test
+
+### Navigate to the TestSys directory
+```bash
+cd ${TESTSYS_DIR}
+```
+
+### Build testing images
+First, the images need to be created and tagged so they can be loaded into a `kind` cluster.
+```bash
+make controller
+make example-test-agent
+docker tag controller controller:demo
+docker tag example-test-agent example-test-agent:demo
+```
+This step uses the Makefile targets for building the controller image and the example test agent image.
+The images then need to be tagged so that the cluster uses loaded images instead of ones from external sources.
+
+### Install the TestSys cli
+Install the `cli` from the repo.
+```bash
+cargo install --path cli
+```
+
+### Create the test yaml file
+The yaml file tells the controller and the test agent how the test should be run.
+```bash
+echo '---
+apiVersion: testsys.bottlerocket.aws/v1
+kind: Test
+metadata:
+  name: hello-bones
+  namespace: testsys-bottlerocket-aws
+spec:
+  agent:
+    name: hello-agent
+    image: "example-test-agent:demo"
+    keepRunning: false
+    configuration:
+      mode: Fast
+      person: Bones the Cat
+      helloCount: 3
+      helloDurationMilliseconds: 500
+  resources: []
+  dependsOn: []' > example_test_agent.yaml
+```
+
+### Create the `kind` cluster
+Check if the `kind` cluster already exists.
+```bash
+kind get clusters
+```
+
+If the cluster already exists, it should be deleted.
+```bash
+kind delete cluster --name testsys-demo
+```
+
+Create the new demo cluster.
+```bash
+kind create cluster --name testsys-demo
+```
+
+Now the images created earlier need to be added to the cluster.
+```bash
+kind load docker-image controller:demo example-test-agent:demo --name testsys-demo
+```
+
+### Install TestSys
+Before any tests can be run, the k8s cluster needs to understand what a test is and how to handle them. 
+To do this TestSys needs to be installed to the cluster.
+```bash
+cli install --controller-uri controller:demo
+```
+
+### Run the test
+The cli installed earlier gives the ability to run tests on a cluster.
```

Comment:
nit: "cli" in this context is an acronym so should be properly capitalized.
```suggestion
The CLI installed earlier gives the ability to run tests on a cluster.
```
and elsewhere as well.